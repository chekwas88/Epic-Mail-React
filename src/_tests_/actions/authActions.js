// import {
//   registerAction, loginAction, clearErrors, processRequest
// } from '../../redux/actions/authActions';

// describe('test for auth actions', () => {
//   test('should setup register action', () => {
//     const action = registerAction({
//       firstName: 'chisom',
//       lastName: 'chekwas',
//       password: '12345678',
//       confirmPassword: '12345678',
//       email: 'ipman@mail.com',
//     });
//     expect(action).toEqual({
//         type: 'REGISTER_USER',
//         payload: {

//         }
//     })
//   });
// });
