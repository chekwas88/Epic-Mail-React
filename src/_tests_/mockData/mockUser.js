export default {
  firstName: 'Klaus',
  lastName: 'Michealson',
  userName: 'hybrid',
  email: 'klaus@epicmail.com',
  password: 'password',
  confirmPassword: 'password',
};
