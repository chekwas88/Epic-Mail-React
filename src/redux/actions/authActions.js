import { post } from 'axios';
import { BASE_URL } from '../../Constants/Constants';
import 'regenerator-runtime';
import {
  REGISTER_USER,
  REGISTER_ERROR,
  LOGIN_USER,
  LOGIN_ERROR,
  PROCESS_REQUEST,
  CLEAR_AUTH_ERROR,
} from '../actionTypes/index';


/**
 * @method registerAction
 * @param {object} userData
 * @param {string} history
 * @returns {object} action object
 */
const registerAction = async (userData, history) => {
  try {
    const registerUser = await post(`${BASE_URL}/auth/signup`, userData);
    const { data } = registerUser.data;
    const { user } = data[0];
    const { token } = data[0];
    const {
      firstname, lastname
    } = user;
    localStorage.setItem('fullname', `${firstname} ${lastname}`);
    localStorage.setItem('userToken', token);
    history.push('/inbox');
    return {
      type: REGISTER_USER,
      payload: { ...user, token }
    };
  } catch (error) {
    return {
      type: REGISTER_ERROR,
      payload: error.response.data,
    };
  }
};

/**
 * @method loginAction
 * @param {object} userData
 *  @param {string} history
 * @returns {object} action object
 */
const loginAction = async (userData, history) => {
  try {
    const loggedUser = await post(`${BASE_URL}/auth/login`, userData);
    const { data } = loggedUser.data;
    const { user } = data[0];
    const { token } = data[0];
    const {
      firstname, lastname
    } = user;
    localStorage.setItem('fullname', `${firstname} ${lastname}`);
    localStorage.setItem('userToken', token);
    history.push('/inbox');
    return {
      type: LOGIN_USER,
      payload: { ...user, token }
    };
  } catch (error) {
    return {
      type: LOGIN_ERROR,
      payload: error.response.data,
    };
  }
};

/**
 * @method clearErrors
 * @returns {object} action object
 */
const clearErrors = () => ({
  type: CLEAR_AUTH_ERROR
});

/**
 * @method processingRequest
 * @returns {object} action object
 */
const processRequest = () => ({
  type: PROCESS_REQUEST
});


export {
  registerAction, loginAction, clearErrors, processRequest,
};
