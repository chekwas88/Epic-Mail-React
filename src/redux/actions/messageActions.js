import { get, post } from 'axios';
import { BASE_URL } from '../../Constants/Constants';
import 'regenerator-runtime';
import {
  GET_RECEIVED_MESSAGES,
  GET_RECEIVED_MESSAGES_ERROR,
  PROCESS_REQUEST,
  SEND_MESSAGE,
  SEND_MESSAGE_ERROR,

} from '../actionTypes/index';


/**
 * @method registerAction
 * @param {object} messageBody
 * @returns {object} action object
 */
const sendMessageAction = async (messageBody) => {
  try {
    const payload = await post(`${BASE_URL}/messages`, messageBody, {
      headers: { Authorization: `Bearer ${localStorage.getItem('userToken')}` }
    });
    const { data } = payload.data;
    const { postedMessage } = data[0].data;

    return {
      type: SEND_MESSAGE,
      payload: { ...postedMessage }
    };
  } catch (error) {
    return {
      type: SEND_MESSAGE_ERROR,
      payload: error.response.data,
    };
  }
};

/**
 * @method getReceivedMessagesAction
 * @returns {object} action object
 */
const getReceivedMessagesAction = async () => {
  try {
    const payload = await get(`${BASE_URL}/messages`, {
      headers: { Authorization: `Bearer ${localStorage.getItem('userToken')}` }
    });
    const { data } = payload.data;
    const inboxMessages = data[0].data;
    return {
      type: GET_RECEIVED_MESSAGES,
      payload: { inboxMessages }
    };
  } catch (error) {
    return {
      type: GET_RECEIVED_MESSAGES_ERROR,
      payload: error.response.data,
    };
  }
};

/**
 * @method getSentMessagesAction
 * @returns {object} action object
 */
const getSentMessagesAction = async () => {
  try {
    const payload = await get(`${BASE_URL}/messages/sent`, {
      headers: { Authorization: `Bearer ${localStorage.getItem('userToken')}` }
    });
    const { data } = payload.data;
    const sentMessages = data[0].data;
    return {
      type: GET_RECEIVED_MESSAGES,
      payload: { sentMessages }
    };
  } catch (error) {
    return {
      type: GET_RECEIVED_MESSAGES_ERROR,
      payload: error.response.data,
    };
  }
};

/**
 * @method processingRequest
 * @returns {object} action object
 */
const processRequest = () => ({
  type: PROCESS_REQUEST
});


export {
  getReceivedMessagesAction, getSentMessagesAction, processRequest, sendMessageAction
};
