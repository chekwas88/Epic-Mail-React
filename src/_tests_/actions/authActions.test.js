import nock from 'nock';
import { registerAction, loginAction } from '../../redux/actions/authActions';
import mockUser from '../mockData/mockUser';

const userData = {
  email: 'klaus@epicmail.com',
  password: 'password',
};

const history = { push: jest.fn() };

describe('authAction', () => {
  afterEach(() => {
    nock.cleanAll();
  });

  it('should register a user', (done) => {
    nock('https://agentcorvinus-epic-mail.herokuapp.com')
      .post('/v1/api/auth/signup')
      .reply(201, {
        message: 'user registration was successful',
        user: {
          id: 24,
          firstName: 'Klaus',
          lastName: 'Michealson',
          email: 'klaus@epicmail.com',
          userName: 'hybrid',
        },
        token:
          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjI0LCJlbWFpbCI6InByaW5jZXNzQGdtYWlsLmNvbSIsInVzZXJOYW1lIjoiZ3IiLCJpYXQiOjE1NjI1Mjg2NTcsImV4cCI6MTU2MzEzMzQ1N30.oWthtPvSh-zz4RwgHZsJtdxpjhHlUKix0oK1I9nqkOA',
      });
    expect(registerAction(mockUser, history)).toMatchSnapshot();
    done();
  });

  it('should return an error when user exits', (done) => {
    const error = 'This username is already in use';
    nock('https://agentcorvinus-epic-mail.herokuapp.com')
      .post('/v1/api/auth/signup')
      .reply(400, { error });
    expect(registerAction({})).toMatchSnapshot();
    done();
  });

  it('Logs in a user', (done) => {
    nock('https://agentcorvinus-epic-mail.herokuapp.com')
      .post('/v1/api/auth/login', { email: 'klaus@epicmail.com', password: 'password' })
      .reply(201, {
        message: 'user registration was successful',
        user: {
          id: 24,
          firstName: 'Klaus',
          lastName: 'Michealson',
          email: 'klaus@epicmail.com',
          userName: 'hybrid',
        },
        token:
          'klwmac.,x;l/,.;l,qwakwehfhioqfhhbncnk'
      });
    expect(loginAction(userData, history)).toMatchSnapshot();
    done();
  });

  it('Logs in a user', (done) => {
    nock('https://agentcorvinus-epic-mail.herokuapp.com')
      .post('/v1/api/auth/login')
      .replyWithError();
    expect(loginAction(userData, history)).toMatchSnapshot();
    done();
  });
});
