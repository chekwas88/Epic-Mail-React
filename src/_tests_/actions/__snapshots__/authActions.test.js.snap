// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authAction Logs in a user 1`] = `Promise {}`;

exports[`authAction Logs in a user 2`] = `Promise {}`;

exports[`authAction should register a user 1`] = `Promise {}`;

exports[`authAction should return an error when user exits 1`] = `Promise {}`;
